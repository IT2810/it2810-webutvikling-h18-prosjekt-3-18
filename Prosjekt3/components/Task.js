import React, { Component } from 'react';
import { View, Text, StyleSheet, FlatList, Button, TextInput } from 'react-native';
import CheckBox from 'react-native-check-box'

class Task extends Component {
    constructor() {
        super();
    }

    render() {
        return (
            <View style={styles.container}>
                <CheckBox
                    isChecked={this.props.checked}
                    onClick={() => {this.props.handleCheckbox(this.props.id)}}
                />
                <Text> {this.props.name} </Text>
            </View>
        )
    }
}

const styles = StyleSheet.create({
    container: {
        flexDirection: 'row',
        flex:0.95,
        marginLeft:2,
        paddingTop:2,
        marginTop:2
    }
});

export default Task;