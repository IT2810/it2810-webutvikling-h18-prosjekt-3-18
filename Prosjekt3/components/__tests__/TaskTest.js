import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import Task from '../../components/Task';

it('renders correctly', () => {
    const tree = renderer.create(<Task>Snapshot test!</Task>).toJSON();
    expect(tree).toMatchSnapshot();
});
