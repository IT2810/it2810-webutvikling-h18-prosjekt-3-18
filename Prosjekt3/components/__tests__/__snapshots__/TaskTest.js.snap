// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "flex": 0.95,
      "flexDirection": "row",
      "marginLeft": 2,
      "marginTop": 2,
      "paddingTop": 2,
    }
  }
>
  <View
    accessible={true}
    isTVSelectable={true}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={null}
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
        }
      }
    >
      <Image
        source={
          Object {
            "testUri": "../../react-native-check-box/img/ic_check_box_outline_blank.png",
          }
        }
        style={
          Object {
            "tintColor": undefined,
          }
        }
      />
    </View>
  </View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  >
     
     
  </Text>
</View>
`;
